<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1065</width>
    <height>764</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_5">
    <item>
     <widget class="GLView" name="openGLWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1065</width>
     <height>26</height>
    </rect>
   </property>
   <property name="nativeMenuBar">
    <bool>false</bool>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextOnly</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="actionWorld_1"/>
   <addaction name="actionWorld_2"/>
   <addaction name="actionWorld_3"/>
   <addaction name="actionWorld_3_Part_2"/>
   <addaction name="actionWorld_4"/>
   <addaction name="actionWorld_5"/>
   <addaction name="actionWorld_6"/>
   <addaction name="actionWorld_7"/>
   <addaction name="actionWorld_8"/>
   <addaction name="actionWorld_9"/>
  </widget>
  <widget class="QDockWidget" name="propertyPanel">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>496</width>
     <height>691</height>
    </size>
   </property>
   <property name="baseSize">
    <size>
     <width>800</width>
     <height>0</height>
    </size>
   </property>
   <property name="features">
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QTabWidget" name="tabWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Ignored" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tabPoints">
        <attribute name="title">
         <string>Points</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QListWidget" name="listPoints">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="dragEnabled">
            <bool>false</bool>
           </property>
           <property name="dragDropOverwriteMode">
            <bool>false</bool>
           </property>
           <property name="dragDropMode">
            <enum>QAbstractItemView::InternalMove</enum>
           </property>
           <property name="movement">
            <enum>QListView::Snap</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QPushButton" name="btnAddPoint">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="text">
               <string>Add</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnDeletePoint">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Delete</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Representation</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBoxRepresentation">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <item>
            <property name="text">
             <string>None</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>ToadHouse Green</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>ToadHouse Yellow</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>ToadHouse Red</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Tower</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Castle</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Final Castle</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>GhostHouse</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Anchor</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Warp Canon</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Peachs Castle</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Fixed</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>10</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QTabWidget" name="tabWidget_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Ignored" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="tabPosition">
            <enum>QTabWidget::North</enum>
           </property>
           <property name="tabShape">
            <enum>QTabWidget::Rounded</enum>
           </property>
           <property name="currentIndex">
            <number>0</number>
           </property>
           <property name="iconSize">
            <size>
             <width>16</width>
             <height>16</height>
            </size>
           </property>
           <property name="elideMode">
            <enum>Qt::ElideNone</enum>
           </property>
           <property name="usesScrollButtons">
            <bool>true</bool>
           </property>
           <property name="documentMode">
            <bool>false</bool>
           </property>
           <property name="tabsClosable">
            <bool>false</bool>
           </property>
           <property name="movable">
            <bool>false</bool>
           </property>
           <property name="tabBarAutoHide">
            <bool>false</bool>
           </property>
           <widget class="QWidget" name="tabAttributes">
            <attribute name="title">
             <string>Attributes</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="QListWidget" name="listEvents">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Ignored" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QWidget" name="widget_2" native="true">
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <item>
                 <widget class="QPushButton" name="btnAddEvent">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Add</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="btnDeleteEvent">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Delete</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tabAnimsOnExit">
            <attribute name="title">
             <string>Exit Anims</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QListWidget" name="listExitAnims">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Ignored" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QWidget" name="widget_4" native="true">
               <layout class="QHBoxLayout" name="horizontalLayout_6">
                <item>
                 <widget class="QPushButton" name="btnAddExitAnim">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Add</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="btnDeleteExitAnim">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Delete</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tabAnimsOnSecretExit">
            <attribute name="title">
             <string>Secret Exit Anims</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QListWidget" name="listSecretExitAnims">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Ignored" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QWidget" name="widget_5" native="true">
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <item>
                 <widget class="QPushButton" name="btnAddSecretExitAnim">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Add</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="btnDeleteSecretExitAnim">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Delete</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tabIndirectConnections">
            <attribute name="title">
             <string>Indirect Connections</string>
            </attribute>
            <layout class="QVBoxLayout" name="verticalLayout_5">
             <item>
              <widget class="QListWidget" name="listIndirectConnections">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Ignored" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QWidget" name="widget_6" native="true">
               <layout class="QHBoxLayout" name="horizontalLayout_8">
                <item>
                 <widget class="QPushButton" name="btnAddUnnconnectedPoint">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Add</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="btnDeleteUnconnectedPoint">
                  <property name="enabled">
                   <bool>false</bool>
                  </property>
                  <property name="text">
                   <string>Delete</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Position</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_3" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="maximumSize">
               <size>
                <width>14</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>x:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="valX">
              <property name="styleSheet">
               <string notr="true">color: rgb(255, 0, 0);</string>
              </property>
              <property name="buttonSymbols">
               <enum>QAbstractSpinBox::UpDownArrows</enum>
              </property>
              <property name="showGroupSeparator" stdset="0">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>-32000.000000000000000</double>
              </property>
              <property name="maximum">
               <double>32000.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_5">
              <property name="maximumSize">
               <size>
                <width>14</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>y:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="valY">
              <property name="styleSheet">
               <string notr="true">color: rgb(0, 94, 255);</string>
              </property>
              <property name="minimum">
               <double>-32000.000000000000000</double>
              </property>
              <property name="maximum">
               <double>32000.000000000000000</double>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_6">
              <property name="maximumSize">
               <size>
                <width>14</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>z:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QDoubleSpinBox" name="valZ">
              <property name="styleSheet">
               <string notr="true">color: rgb(0, 255, 0);</string>
              </property>
              <property name="minimum">
               <double>-32000.000000000000000</double>
              </property>
              <property name="maximum">
               <double>32000.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tabPaths">
        <attribute name="title">
         <string>Paths</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QListWidget" name="listPaths">
           <property name="dragDropMode">
            <enum>QAbstractItemView::InternalMove</enum>
           </property>
           <property name="movement">
            <enum>QListView::Snap</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widget_7" native="true">
           <layout class="QHBoxLayout" name="horizontalLayout_9">
            <item>
             <widget class="QPushButton" name="btnAddPath">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Add</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnDeletePath">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>Delete</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="ShapeView" name="shapeView" native="true">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="autoFillBackground">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Animation</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="valPathAnimation">
           <property name="enabled">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Meta+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Meta+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Extract CSVs</string>
   </property>
   <property name="shortcut">
    <string>Meta+S</string>
   </property>
  </action>
  <action name="actionWorld_1">
   <property name="icon">
    <iconset resource="resources.qrc">
     <normaloff>:/res/images/icon.png</normaloff>:/res/images/icon.png</iconset>
   </property>
   <property name="text">
    <string>World 1</string>
   </property>
  </action>
  <action name="actionWorld_2">
   <property name="icon">
    <iconset>
     <normalon>:/res/images/w2.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>World 2</string>
   </property>
  </action>
  <action name="actionWorld_3">
   <property name="icon">
    <iconset>
     <activeon>:/res/images/w3.png</activeon>
    </iconset>
   </property>
   <property name="text">
    <string>World 3 Part 1</string>
   </property>
  </action>
  <action name="actionWorld_3_Part_2">
   <property name="text">
    <string>World 3 Part 2</string>
   </property>
  </action>
  <action name="actionWorld_4">
   <property name="text">
    <string>World 4</string>
   </property>
  </action>
  <action name="actionWorld_5">
   <property name="text">
    <string>World 5</string>
   </property>
  </action>
  <action name="actionWorld_6">
   <property name="text">
    <string>World 6</string>
   </property>
  </action>
  <action name="actionWorld_7">
   <property name="text">
    <string>World 7</string>
   </property>
  </action>
  <action name="actionWorld_8">
   <property name="text">
    <string>World 8</string>
   </property>
  </action>
  <action name="actionWorld_9">
   <property name="text">
    <string>World 9</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLView</class>
   <extends>QOpenGLWidget</extends>
   <header>glview.h</header>
  </customwidget>
  <customwidget>
   <class>ShapeView</class>
   <extends>QWidget</extends>
   <header>shapeview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
